src/server/cmd/pachctl/shell/completions.go

```diff
@@ -15,16 +18,53 @@ import (
 	pps_pretty "github.com/pachyderm/pachyderm/src/server/pps/pretty"
 
 	prompt "github.com/c-bata/go-prompt"
-	"github.com/docker/go-units"
+	units "github.com/docker/go-units"
 )
 
-// RepoCompletion completes repo parameters of the form <repo>
-func RepoCompletion(_, text string, maxCompletions int64) []prompt.Suggest {
-	c, err := client.NewOnUserMachine("user-completion")
-	if err != nil {
-		log.Fatal(err)
+type part int
+
+const (
+	repo part = iota
+	commitOrBranch
+	file
```

Comment:
These are pretty generic names for enum literals, maybe consider using `repoPart`, `commitOrBranchPart`, `filePart` to make it more obvious and less likely to collide?

Reply:
I notice you already have a function called `filePart`, but this might be better as `getPart` or something.

Reply:
Done, went with `parsePart`